### PR TITLE
Remove used messages tracking in the PortHandler

### PR DESCRIPTION
**Pull request by samuelmanzanera**
# Description

This fix reduce the memory footprint by pruning used messages when communicating with a port (external program)

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by apoorv-2204** [comment]
unused?

File: lib/archethic/utils/port_handler.ex [hunk 1]
```diff
@@ -49,11 +49,11 @@ defmodule Archethic.Utils.PortHandler do
         {_port, {:data, <<request_id::32, response::binary>>}},
         state = %{awaiting: awaiting}
       ) do
```
Comment (apoorv-2204) [inline lib/archethic/utils/port_handler.ex:51]: [port_handler.ex], is this for NIF?

Comment (samuelmanzanera) [reply lib/archethic/utils/port_handler.ex:51]: no it's using external program

**Comment by samuelmanzanera** [comment]
> unused?

messages received by the port but not removing from the genserver for their tracking, hence the memory was rising.

File: lib/archethic/utils/port_handler.ex (hunk 1 above)
Comment (apoorv-2204) [reply lib/archethic/utils/port_handler.ex:51]: like hyper_geometric_distribution.c ?

Comment (samuelmanzanera) [reply lib/archethic/utils/port_handler.ex:51]: More like libsodium or tpm

**Review by Neylix — APPROVED** [review]

**Pull request merged**